Gems/PhysX/Code/Source/EditorShapeColliderComponent.h

```diff
@@ -151,6 +159,9 @@ namespace PhysX
         //! @note 16 is the number of subdivisions in the debug cylinder that is loaded as a mesh (not generated procedurally)
         AZ::u8 m_subdivisionCount = 16; 
         mutable GeometryCache m_geometryCache; //!< Cached data for generating sample points inside the attached shape.
+        AZStd::optional<bool> m_previousIsTrigger; //!< Stores the previous trigger setting if the shape is changed to one which does not support triggers.
```

Comment:
question: `optional<bool>` is always a little worrying due to the implicit conversions that _can_ happen, might be worth considering an `enum` to track the state instead? There may be other alternatives or you an leave as-is, it just has the potential to be confusing.

Reply:
Oh yeah, that's a good point. I think I'm always explicit with how I use it here, and its use is pretty limited, so I think it's safe to leave it in this case.